### ISSUE TITLE
Switch version shenanigans (AQUARIUM.)

### ISSUE THREAD

**Issue by Inui-senpai** [koukdw/Aquarium_tools#7]
Greetings. 🙋‍♀️
First of all, thank you for your import/export tools, Russian translation of the Aquarium VN would be impossible without them.

And while I was doing it, I ran into the following problems:
1. If you make a translation for the updated version (which includes some fixes and the Extra Story), you **must** use the Update folder in your patch dir, otherwise your modified files would be ignored by the game (at best) or the game would be crashed into the emulator's main menu (at worst);
2. Modified files with the cyrillic symbols will also be ignored by the game (not sure if the problem is with the game itself or the importer script);
3. Also you should use a specific range of symbols (latin, kanji, etc.), **not** mixing them. The latter will lead to the following problems:
- The game will freeze if the string contains the "†" and "♪" characters (although I didn't catch the freeze right away with the last character);
- The game will also freeze when trying to open the History screen if the choice strings contains the mixed symbols or if you translated the names and left out the Japanese brackets ("【", "】").

I used the version I was downloaded on July, 16 2024

**Comment by kdw-code** [comment]
1. Thanks for the advice, indeed the `Update` folder have priority, you can add your own folder too by changing `packlist.dat`.
2. My tool just export strings and reimport them. 
It doesn't fix the problems with fonts (if the original font doesnt have cyrillic it won't show cyrillic)
Two days ago i've made a tool to export `dat` files to `csv`.  
~~There is a file called `fontresource.dat`, you should try to modify the `font_file`/`font_name` with a font that contains `cyrillic` like the one from `Aquarium PC` (i'm assuming it does since you didn't have problem with it, don't forget to put your font in the system folder too)~~ They are the same font file and they contain cyrillic.
**Ah and don't use google sheets to open the csv, because it remove quotes which will cause problem when reimporting. Instead use a proper csv editor that keep the quotes, there's a vscode extension that can do that i don't remember the name.**
3. This, i don't have any knowledge about this i can investigate, i know `system.dat` have 2 slot with those  "【", "】" characters

Now if the font modification doesn't work then i don't know. There's another font format with extension .fnt
https://github.com/Yggdrasill-Moe/Niflheim/tree/master/NeXAS this repo seems to have tool for it but it's for shift jis so you might need to modify it

I'm investigating the issue right now i'll give you my answer later.

**Comment by kdw-code** [comment]
@Inui-senpai What do you mean by cyrillic symbols ? All cyrillic glyph or special punctuation characters ?
Because so far i can use cyrillic in my mod for the switch game.
Also does it not work on pc too ?

**Comment by Inui-senpai** [comment]
I meant that if you put cyrillic glyphs "as-is", the game won't load the modified file (unlike the PC version), so I just replaced the latin ones in mine font file. Should I retry with the updated tools? 🤔

**Comment by kdw-code** [comment]
I also tried using Kanji in a sentence full of cyrillic character and it works. I didn't change the font tho.
I did the same with the character that are supposed to crash the game but again no crash.

**Comment by koukdw** [comment]
![image](https://github.com/user-attachments/assets/d5e3bfa7-b54c-441c-aff2-95f7a2df34a5)

**Comment by kdw-code** [comment]
![image](https://github.com/user-attachments/assets/8a6ab69c-52e1-4c42-bbd7-bd8fc98c82d6)

**Comment by Inui-senpai** [comment]
Interesting. And what about the choices (you can skip to the first choice section by pressing ZR)? 🤔

**Comment by kdw-code** [comment]
@Inui-senpai 
![image](https://github.com/user-attachments/assets/acebb0af-e275-44b3-8b86-8e2d86f8b346)

**Comment by kdw-code** [comment]
![image](https://github.com/user-attachments/assets/e6e3a1ad-65a7-41a2-9d91-2b61de8419bd)

**Comment by Inui-senpai** [comment]
Mhm... And if you update the game to the ver. 1.0.4, it will work the same way? 🤔

**Comment by kdw-code** [comment]
Yes, it works the same. I think it has to do with the font file you replaced.

**Comment by Inui-senpai** [comment]
That's strange. I also tried without the font replacement, and I've had experienced the same troubles. Moreover, I've ran into these troubles not only on yuzu EA, but also on Ryujinx (and there was also someone who ran into these troubles on the original console: https://vk.com/topic-165582702_39071884?post=11564). Could it be because I've dumped all game's files via Yanu, or is it irrelevant? 🤔

**Comment by kdw-code** [comment]
I mean if the file were dumped incorrectly it wouldn't even load because the engine would not be able to read the file format.
Please share a file that you know doesn't work (preferrably near the start of the game) so i can analyze it.
Also share the font modification.

**Comment by Inui-senpai** [comment]
Just now found some free time for testing, sorry for keep you waiting. I used the same old version of tools (and the same yuzu EA 4176), and now I'm really confused: the fully translated files with the cyrillic glyphs put "as-is" works as usual, both with the original and replaced font. Why didn't it work before? 😵‍💫

Original font:
![0100d11018a7e000_2024-12-12_05-35-22-815](https://github.com/user-attachments/assets/f785aa8e-803a-4d89-b7d1-c4bab72b19bb)
![0100d11018a7e000_2024-12-12_05-38-24-535](https://github.com/user-attachments/assets/7978a5f3-74d4-4c49-8118-e157f20548d2)
Replaced font (I also replaced the `fontresource` and `system` files to replace the "friendly" font name for the game):
![0100d11018a7e000_2024-12-12_05-44-20-378](https://github.com/user-attachments/assets/3718ed59-98c0-4ea4-aa50-07c1f6eaaa58)
![0100d11018a7e000_2024-12-12_05-44-58-447](https://github.com/user-attachments/assets/c151a4fd-bada-4d8e-a62b-d3171b0890c6)

Here's the files I used for the patch, just in case:
[aqu_cyr_test.zip](https://github.com/user-attachments/files/18103382/aqu_cyr_test.zip)

But anyway, the cyrillic glyphs has spaces between them, even in the font that doesn't have it, but I have to edit the `main` file in the ExeFS in order to fix this (and to translate the system tooltips), as far as I understand. 🤔

**Comment by Inui-senpai** [comment]
Half-width katakana as cyrillic (aka "the unstable one") hack: the game will hang if you skip to the choice section and try to open the Backlog; but if you remove the modified font and relevant files, the hang will occur earlier (when the game attempts to show name in brackets in the Backlog). Also it has a problem with the "Ё" and "Й" glyphs.

Patch:
[aqu_rus_unstable.zip](https://github.com/user-attachments/files/18129724/aqu_rus_unstable.zip)

Screenshots:
![image](https://github.com/user-attachments/assets/3d188b61-4fac-40d9-b3bf-3857f2756a7b)
![image](https://github.com/user-attachments/assets/be4a1cb6-e670-4f1a-a5db-6db0aa27241f)

**Comment by Inui-senpai** [comment]
By the way...
> I have to edit the `main` file in the ExeFS in order to fix this (and to translate the system tooltips), as far as I understand

I tried to remove the "Made choice" caption from the Backlog through wxmEdit (I chose the UTF-8 encoding in it), and not only did it not disappear, but the system message about the skipping to the next choice was stripped to the one symbol. What am I doing wrong? 🤔

**Comment by kdw-code** [comment]
> But anyway, the cyrillic glyphs has spaces between them, even in the font that doesn't have it, but I have to edit the main file in the ExeFS in order to fix this (and to translate the system tooltips), as far as I understand. 🤔

I don't know how to achieve this, the fact that it's a switch game make it more complicated to me, i heard of skyline to hook switch game (does that even work on emulator idk)

**Comment by Inui-senpai** [comment]
I think we can also consider this option: https://github.com/shadowninja108/exlaunch

It works on yuzu, but cause crashes on Ryujinx, as far as I know. But I don't use the latter because of the RTC problems, so it's not that critical, I guess. 🤔
